src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -82,13 +91,28 @@ public static function fromDsn(string $dsn, array $options = array(), bool $debu
         }
 
         $exchangeOptions = $amqpOptions['exchange'];
-        $queueOptions = $amqpOptions['queue'];
+        $queueOptions = self::getQueueOptions($amqpOptions['queue']);
 
         unset($amqpOptions['queue'], $amqpOptions['exchange']);
 
         return new self($amqpOptions, $exchangeOptions, $queueOptions, $debug, $amqpFactory);
     }
 
+    private static function getQueueOptions(array $queueOptions): array
+    {
+        if (!\is_array($queueOptions['arguments'] ?? false)) {
+            return $queueOptions;
+        }
+
+        foreach (self::ARGUMENTS_AS_INTEGER as $key) {
+            if (array_key_exists($key, $queueOptions['arguments'])) {
```

Comment:
Do we really wants that when an argument is `null` the value becomes zero instead of unset the key?

Reply:
Fixed, an exception is now thrown if the value is not an integer